presto-druid/src/main/java/com/facebook/presto/druid/DruidQueryGeneratorContext.java

```diff
@@ -104,7 +105,12 @@ private DruidQueryGeneratorContext(
 
     public DruidQueryGeneratorContext withFilter(String filter)
     {
-        checkState(!hasFilter(), "Druid doesn't support filters at multiple levels");
+        if (!hasAggregation()) {
+            checkState(!hasFilter(), "Druid doesn't support filters at multiple levels under AggregationNode");
+        }
+        else {
+            throw new PrestoException(DRUID_PUSHDOWN_UNSUPPORTED_EXPRESSION, "Druid does not support filter on top of AggregationNode.");
```

Comment:
I think better not through PrestoException here. Could we do:
checkState(!hasAggregation && !hasFilter, ...)

Reply:
Throw PrestoException here is intended here I think. checkState will make query failed directly. if query are like:
select a , sum(b) from druid_table where c = 'xxx' group by 1 having sum(b) > 0;
Throw PrestoException will help make the filter sum(b) > 0(on top of aggregation node) executed in presto(not in druid). The reason not push down such filter right now is because we could not make sure all filters on top of aggregation node could be pushed down. Following code will catch this exception and make queries work(DruidQueryGenerator):
    public Optional<DruidQueryGeneratorResult> generate(PlanNode plan, ConnectorSession session)
    {
        try {
            DruidQueryGeneratorContext context = requireNonNull(plan.accept(
                    new DruidQueryPlanVisitor(session),
                    new DruidQueryGeneratorContext()),
                    "Resulting context is null");
            return Optional.of(new DruidQueryGeneratorResult(context.toQuery(), context));
        }
        catch (PrestoException e) {
            log.debug(e, "Possibly benign error when pushing plan into scan node %s", plan);
            return Optional.empty();
        }
    }

Reply:
If any code I need to revert back to be compatible with #14952 , just tell me.